test/services/MovementMessageOrchestratorServiceSpec.scala

```diff
@@ -0,0 +1,140 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import base.SpecBase
+import generators.ModelGenerators
+import models.ArrivalStatus.GoodsReleased
+import models.Arrival
+import models.ArrivalId
+import models.BoxId
+import models.GoodsReleasedResponse
+import models.InboundMessageRequest
+import models.MessageSender
+import models.MovementMessageWithoutStatus
+import models.SubmissionState
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.refEq
+import org.mockito.Mockito.when
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.inject.bind
+import play.api.mvc.Headers
+import play.api.test.Helpers.running
+
+import scala.concurrent.Future
+
+class MovementMessageOrchestratorServiceSpec extends SpecBase with ModelGenerators with ScalaCheckDrivenPropertyChecks {
+
+  val mockInboundRequestService       = mock[InboundRequestService]
+  val mockSaveMessageService          = mock[SaveMessageService]
+  val mockPushPullNotificationService = mock[PushPullNotificationService]
+
+  "MovementMessageOrchestratorService" - {
+
+    "saveNCTSMessage" - {
+
+      "must return a InboundMessageRequest when successfully processing a request xml" in {
+
+        val arrival       = arbitrary[Arrival].sample.value
+        val message       = arbitrary[MovementMessageWithoutStatus].sample.value
+        val messageSender = MessageSender(arrival.arrivalId, arrival.nextMessageCorrelationId)
+        val xml           = <test>TestXml</test>
+
+        val expectedResult = InboundMessageRequest(arrival, GoodsReleased, GoodsReleasedResponse, message)
+
+        when(mockInboundRequestService.makeInboundRequest(any(), any(), any())(any())).thenReturn(Future.successful(Right(expectedResult)))
+        when(mockSaveMessageService.saveInboundMessage(any(), any())(any())).thenReturn(Future.successful(Right(())))
+        when(mockPushPullNotificationService.sendPushNotification(any(), any())(any())).thenReturn(Future.unit)
+
+        val application = baseApplicationBuilder
+          .overrides(bind[InboundRequestService].toInstance(mockInboundRequestService))
+          .overrides(bind[SaveMessageService].toInstance(mockSaveMessageService))
+          .overrides(bind[PushPullNotificationService].toInstance(mockPushPullNotificationService))
+          .build()
+
+        running(application) {
```

Comment:
I think we should avoid running up a Guice apps for every test step as much as we can - I have seen this approach take test suites into tens of minutes just due to the overhead of starting apps. 
I don't think you need to run the app unless you want to hit a controller endpoint via the router - the rest of the time you can declare the application once and then use the `app.injector` to get things.